src/Crypto/ECC/Edwards25519.hs

```diff
@@ -50,13 +53,15 @@ import qualified Data.ByteArray as B hiding (append)
 
 -- | Represent a scalar in the base field
 newtype Scalar = Scalar { unScalar :: ByteString }
+    deriving (Eq, Ord, NFData, Hashable)
```

Comment:
same as the wallet. Eq and Ord are security issue, possibly Hashable _could_ be too, although I think at the moment it's not.